fuel-vm/src/interpreter/gas.rs

```diff
@@ -21,37 +27,102 @@ impl<S, Tx> Interpreter<S, Tx> {
     }
 
     pub(crate) fn dependent_gas_charge(&mut self, gas_cost: DependentCost, arg: Word) -> Result<(), RuntimeError> {
-        if gas_cost.dep_per_unit == 0 {
-            self.gas_charge(gas_cost.base)
-        } else {
-            self.gas_charge(gas_cost.base.saturating_add(arg.saturating_div(gas_cost.dep_per_unit)))
-        }
+        let current_contract = self.contract_id();
+        let ReadRegisters { pc, ggas, cgas, is, .. } = split_registers(&mut self.registers).0;
+        let profiler = ProfileGas {
+            pc: pc.as_ref(),
+            is: is.as_ref(),
+            current_contract,
+            profiler: &mut self.profiler,
+        };
+        dependent_gas_charge(cgas, ggas, profiler, gas_cost, arg)
     }
 
     pub(crate) fn gas_charge(&mut self, gas: Word) -> Result<(), RuntimeError> {
+        let current_contract = self.contract_id();
+        let ReadRegisters { pc, ggas, cgas, is, .. } = split_registers(&mut self.registers).0;
+
+        let profiler = ProfileGas {
+            pc: pc.as_ref(),
+            is: is.as_ref(),
+            current_contract,
+            profiler: &mut self.profiler,
+        };
+        gas_charge(cgas, ggas, profiler, gas)
+    }
+}
+
+pub(crate) fn dependent_gas_charge(
+    mut cgas: RegMut<CGAS>,
+    ggas: RegMut<GGAS>,
+    mut profiler: ProfileGas<'_>,
+    gas_cost: DependentCost,
+    arg: Word,
+) -> Result<(), RuntimeError> {
+    if gas_cost.dep_per_unit == 0 {
+        gas_charge(cgas, ggas, profiler, gas_cost.base)
+    } else {
+        let cost = dependent_gas_charge_inner(cgas.as_mut(), ggas, gas_cost, arg)?;
+        profiler.profile(cgas.as_ref(), cost);
```

Comment:
Do we need to put `profiler` under the `#[cfg(feature = "profile-gas")]` flag?

Reply:
No this is a blank profiler if the flag is not set